app/src/main/java/com/example/campusguide/directions/TransitRouteAdapter.kt

```diff
@@ -0,0 +1,142 @@
+package com.example.campusguide.directions
+
+import android.app.Activity
+import android.graphics.Color
+import android.graphics.Typeface
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.example.campusguide.R
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+class TransitRouteAdapter constructor(private val activity: Activity) : BaseAdapter() {
+
+    private val results: ArrayList<TransitRoute> = arrayListOf()
+
+    /**
+     * Get a View that displays the data at the specified position in the data set.
+     */
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+
+        val inflater = LayoutInflater.from(activity)
+        val row = inflater.inflate(R.layout.transit_route_item, parent, false)
+
+        // Set the route title
+        row.findViewById<TextView>(R.id.routeTitle).text = results[position].title
+
+        // Set the route travel time
+        val travelTime = "${results[position].duration / 60} min"
+        row.findViewById<TextView>(R.id.routeDuration).text = travelTime
+
+        // Display the first four steps
```

Comment:
Why the first 4 steps? Cause it looks nice?

Reply:
Just so it fits nicely into one line, there's really no point in displaying more steps
I also, admittedly, looked at how Google Maps handled the case of a route containing a lot of steps